### PR TITLE
Fix starter.yaml for statusreconciler, apiversion extensions/v1beta1.

### PR DESCRIPTION
**Pull request by arturo-c**
Fixes https://github.com/kubernetes/test-infra/issues/12354

### CODE REVIEW

File: prow/cluster/starter.yaml [hunk 1]
```diff
@@ -434,7 +434,7 @@ spec:
           serviceName: hook
           servicePort: 8888
 ---
-apiVersion: apps/v1
+apiVersion: extensions/v1beta1
```
Comment (stevekuznetsov) [inline prow/cluster/starter.yaml]: Do we have a recommended version of k8s to use? This is not necessary on recent releases AFAICT

Comment (arturo-c) [reply prow/cluster/starter.yaml]: not completely sure, I just noticed this same apiVersion applied to the rest of the deployments in this yaml, so I'm thinking whatever version is used it should be the same in all within this starter.yaml?

Comment (stevekuznetsov) [reply prow/cluster/starter.yaml]: Yeah, that makes sense

**Pull request merged**